.github/workflows/pipeline.yml

```diff
@@ -1,36 +1,28 @@
-name: lint and test
+name: Install and test
 
 on: [push]
 
 jobs:
-  lint_and_test:
+  install_and_test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.9", "3.10", "3.11", "3.12" ]
 
     steps:
-      - uses: actions/checkout@v2
-      - name: Set up Python 3.8
-        uses: actions/setup-python@v2
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
         with:
-          python-version: 3.8
+          python-version: ${{ matrix.python-version }}
+      - name: Display Python version
+        run: python -c "import sys; print(sys.version)"
 
       - name: Install poetry
-        run: pip install poetry==1.1.12
+        run: pip install poetry>=1.7.1
 
       - name: Install code
         run: poetry install
 
-      - name: Lint with flake8
-        run: |
-          # stop the build if there are Python syntax errors or undefined names
-          poetry run flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
-          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
-          poetry run flake8 . --count --exit-zero --statistics
-
-      - name: Check formatting with Black
-        run: poetry run black . --check
-
-      - name: Check import order with isort
-        run: poetry run isort -c spatial_kde
-
```

Comment:
Why were these steps removed?

Reply:
I just moved them to python-app.yml and .flake8 file.

Reply:
It should be compatible with ci now. I have deleted python-app.yml.